ios/RNSScreen.m

```diff
@@ -295,6 +297,63 @@ - (instancetype)initWithView:(UIView *)view
   return self;
 }
 
+- (UIStatusBarStyle)preferredStatusBarStyle
+{
+  UIViewController *child = [[self childViewControllers] lastObject];
+  if ([child isKindOfClass:[RNScreensNavigationController class]] || [child isKindOfClass:[RNScreensViewController class]]) {
+    return child.preferredStatusBarStyle;
+  }
+  RNSScreenStackHeaderConfig *config = [self findConfigForScreen];
+  return config && config.statusBarStyle ? [self statusBarStyleForRNSStatusBarStyle:config.statusBarStyle] : [self statusBarStyleForRNSStatusBarStyle:RNSStatusBarStyleAuto];
+}
+
+- (UIStatusBarAnimation)preferredStatusBarUpdateAnimation
+{
+  UIViewController *child = [[self childViewControllers] lastObject];
+  if ([child isKindOfClass:[RNScreensNavigationController class]] || [child isKindOfClass:[RNScreensViewController class]]) {
+    return child.preferredStatusBarUpdateAnimation;
+  }
+  RNSScreenStackHeaderConfig *config = [self findConfigForScreen];
+  return config && config.statusBarAnimation ? config.statusBarAnimation : UIStatusBarAnimationFade;
+}
+
+- (BOOL)prefersStatusBarHidden
+{
+  UIViewController *child = [[self childViewControllers] lastObject];
+  if ([child isKindOfClass:[RNScreensNavigationController class]] || [child isKindOfClass:[RNScreensViewController class]]) {
+    if ([child childViewControllerForStatusBarHidden]) {
+      return [child childViewControllerForStatusBarHidden].prefersStatusBarHidden;
+    }
+  }
+  RNSScreenStackHeaderConfig *config = [self findConfigForScreen];
+  return config && config.statusBarHidden ? config.statusBarHidden : NO;
+}
+
+- (RNSScreenStackHeaderConfig *)findConfigForScreen
+{
+  // if there is no child navigator and the parent is `RNSScreenContainer`, we should fallback to the parent's (that is not `RNSScreenContainer`) option
+  UIViewController *parent = [self parentViewController];
+  while ([parent isKindOfClass:[RNScreensViewController class]] || [parent isKindOfClass:[RNSScreen class]]) {
+    if ([parent parentViewController] == nil && [parent isKindOfClass:[RNSScreen class]]) {
+      // we are at the top of hierarchy and the controller is RNSScreen so we are in modal screen
+      break;
+    } else {
+      parent = [parent parentViewController];
+    }
+  }
+  RNSScreenView *screenView = [parent isKindOfClass:[RNScreensNavigationController class]]
```

Comment:
If it is not, we will go to the `else` section of the second ternary operator, where we just return the `self.view`, which in the case of `RNSScreen` is always `RNSScreenView`. It probably is good to check if the child of `RNScreensNavigationController` is `RNSScreen` for sure and add comments about the other options.

Reply:
What if instead of doing this parent traversal we eliminate the need of accessing the config in childViewControllerForStatusBarStyle and others? It looks like it'd be much easier to do the up-to-bottom traversal when we're asked to provide child component as opposed to always returning some child VC and in that child VC doing a reverse traversal to find a parent where config is set